shared/services/config/alertmanager-config.go

```diff
@@ -0,0 +1,132 @@
+
+	// The Discord webhook URL for alert notifications
+	DiscordWebhookURL config.Parameter `yaml:"discordWebhookURL,omitempty"`
+}
+
+func NewAlertmanagerConfig(cfg *RocketPoolConfig) *AlertmanagerConfig {
+
+	return &AlertmanagerConfig{
+		Title: "Alertmanager Settings",
+
+		Port: config.Parameter{
+			ID:                 "port",
+			Name:               "Alertmanager Port",
+			Description:        "The port Alertmanager will listen on.",
+			Type:               config.ParameterType_Uint16,
+			Default:            map[config.Network]interface{}{config.Network_All: defaultAlertmanagerPort},
+			AffectsContainers:  []config.ContainerID{config.ContainerID_Alertmanager,config.ContainerID_Prometheus},
+			CanBeBlank:         true,
+			OverwriteOnUpgrade: false,
+		},
+
+		OpenPort: config.Parameter{
+			ID:                 "openPort",
+			Name:               "Expose Alertmanager Port",
+			Description:        "Expose the Alertmanager's port to other processes on your machine, or to your local network so other machines can access it too.",
+			Type:               config.ParameterType_Choice,
+			Default:            map[config.Network]interface{}{config.Network_All: defaultAlertmanagerOpenPort},
+			AffectsContainers:  []config.ContainerID{config.ContainerID_Alertmanager},
+			CanBeBlank:         false,
+			OverwriteOnUpgrade: false,
+			Options:            config.PortModes(""),
+		},
+
+		ContainerTag: config.Parameter{
+			ID:                 "containerTag",
+			Name:               "Alertmanager Container Tag",
+			Description:        "The tag name of the Alertmanager container you want to use on Docker Hub.",
+			Type:               config.ParameterType_String,
+			Default:            map[config.Network]interface{}{config.Network_All: alertmanagerTag},
+			AffectsContainers:  []config.ContainerID{config.ContainerID_Alertmanager},
+			CanBeBlank:         false,
+			OverwriteOnUpgrade: true,
+		},
+
+		DiscordWebhookURL: config.Parameter{
+			ID:                 "discordWebhookURL",
+			Name:               "Alertmanager Discord Webhook URL",
+			Description:        "Discord notifications are sent via the Discord webhook API. See Discord's 'Intro to Webhooks' article to learn how to configure a webhook integration for a channel at https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks",
+			Type:               config.ParameterType_String,
+			Default:            map[config.Network]interface{}{config.Network_All: ""},
+			AffectsContainers:  []config.ContainerID{config.ContainerID_Alertmanager},
+			CanBeBlank:         true,
+			OverwriteOnUpgrade: false,
+		},
+	}
+}
+
+func (cfg *AlertmanagerConfig) GetParameters() []*config.Parameter {
+	return []*config.Parameter{
+		&cfg.Port,
+		&cfg.OpenPort,
+		&cfg.ContainerTag,
+		&cfg.DiscordWebhookURL,
+	}
+}
+
+func (cfg *AlertmanagerConfig) GetConfigTitle() string {
+	return cfg.Title
+}
+
+// Used by text/template to format alertmanager.yml
```

Comment:
`GetOpenPorts` and `UpdateConfigurationFile` have a very similar implementation to equivelents elsewhere but I felt the code was cleaner and better factored by putting them here instead of sprinkling the "alerting concern" throughout the codebase so much.

Reply:
it's on my mental todo list to move the text/template specific functions closer to the values they scan.